### PR TITLE
Engine setup

### PR DESCRIPTION
**Pull request by mvandenbeuken**
Basic engine configuration:
- Added simplecov and set min coverage at 100%.
- Added rspec
- Added combustion for engine testing
  Dynamically mounts the engine and sets up the asset precompile list when the gem is required. This allows us to mix in the appropriate stuff in test and/or development environments.
  CreditCardsController:
- acts like the Moneris hosted tokenization iframe content
- serves up a credit card input
- serves up javascript that handles postMessage events, and returns the same data that Moneris does:
  - dataKey: a token representing the credit card
  - bin: the first 6 digits of the credit card
  - responseCode: 1 if the credit card is in the valid credit card list, 51 otherwise
    Javascript is contained in the MonerisSimulator namespace.
    Used styles applied to our Moneris input.

:elephant:

### CODE REVIEW

File: app/assets/stylesheets/moneris_simulator/inputs.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
+body{
+  margin: 0;
+}
+
+.monerisInput{
+	display: block;
```
Comment (rubene) [inline app/assets/stylesheets/moneris_simulator/inputs.scss]: tabs

File: app/assets/stylesheets/moneris_simulator/inputs.scss [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  text-indent: 0px;
+  text-shadow: none;
+  text-transform: none;
+  transition-delay: 0s, 0s;
+  transition-duration: 0.45s, 0.45s;
+  transition-property: box-shadow, border-color;
+  transition-timing-function: ease, ease-in-out;
+  width: 100%;
+  word-spacing: 0px;
+  writing-mode: lr-tb;
```
Comment (rubene) [inline app/assets/stylesheets/moneris_simulator/inputs.scss:69]: malware

File: spec/internal/config/database.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+test:
+  adapter:  sqlite3
```
Comment (rubene) [inline spec/internal/config/database.yml:2]: change this to pg?

File: spec/internal/db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+ActiveRecord::Schema.define do
+  #
```
Comment (rubene) [inline spec/internal/db/schema.rb]: comment

File: lib/moneris_simulator/engine.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+module MonerisSimulator
+  class Engine < ::Rails::Engine
+    isolate_namespace MonerisSimulator
+
+    config.after_initialize do |app|
```
Comment (rubene) [inline lib/moneris_simulator/engine.rb:5]: :+1:

File: app/assets/stylesheets/moneris_simulator/inputs.scss (hunk 2 above)
Comment (mvandenbeuken) [inline app/assets/stylesheets/moneris_simulator/inputs.scss:69]: lol. red code!!!

File: spec/internal/config/database.yml (hunk 3 above)
Comment (mvandenbeuken) [inline spec/internal/config/database.yml:2]: Figured I'd keep it lightweight here. No models in this engine.

**Comment by mvandenbeuken** [comment]
:wave:

**Comment by rubene** [comment]
:elephant: :elephant: :elephant:

File: app/views/moneris_simulator/credit_cards/new.html.erb [hunk 6]
```diff
@@ -0,0 +1 @@
+<input type="tel" class="monerisInput" id="monerisDataInput" autocomplete="cc-number" />
```
Comment (jbabe) [inline app/views/moneris_simulator/credit_cards/new.html.erb:1]: could you use `type=number` here, or does moneris use `type=tel` on their form?

File: app/views/layouts/moneris_simulator/application.html.erb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<html>
+  <head>
+    <%= stylesheet_link_tag    "moneris_simulator", media: "all" %>
+    <%= javascript_include_tag "moneris_simulator" %>
```
Comment (jbabe) [inline app/views/layouts/moneris_simulator/application.html.erb:4]: Do we want to exclude JS to make it closer to the moneris page?

File: spec/rails_helper.rb [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+require "rails/all"
+require "rspec/rails"
+require "factory_girl_rails"
+require "webmock/rspec"
+require "database_cleaner"
+require "sass-rails"
+require "coffee-rails"
+require "pry"
+
+WebMock.disable_net_connect! allow_localhost: true
```
Comment (jbabe) [inline spec/rails_helper.rb:23]: :+1:

File: app/assets/javascripts/moneris_simulator/credit_card.coffee [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    @creditCardNumber().substring 0, 6
+
+  responseCode: () ->
+    if @validCreditCard() then "1" else "51"
+
+  validCreditCard: () ->
+    @creditCardNumber() in @validCreditCards
+
+  defaults: () ->
+    validCreditCards: ["4502285070000007", "4924190000003750", "4551210032790206", "4924190000000145", "4524000000015080", \
```
Comment (jbabe) [inline app/assets/javascripts/moneris_simulator/credit_card.coffee:31]: are any of these AMEX? Do we want to generate tokens for AMEX cards too? I think the moneris iframe does, correct ( and fails when you tried to use the token if you don't support the card type)?

**Pull request merged**